### PR TITLE
Fix url replace error COUNTRY=cambodia

### LINKED ISSUE #552: Kobo forms - choosing a new date fails to load data
If a user chooses a WMS layers followed by a kobo layer, then deselects the WMS layer, date selection no longer works on the kobo layer.

Steps to reproduce:
1. On master, REACT_APP_COUNTRY=cambodia
2. Activate layer Rainfall amount > Rainfall aggregate 10-day
3. Activate layer Exposure > Flood report
4. De-activate rainfall aggregate layer
5. From the date picker, choose another date. The URL correctly updates, but no request is sent and no data is loaded

All ok through step 3:
Screen Shot 2022-08-29 at 18 12 42

This is what I see for the date picker after de-activating the WMS layer (step 4):
Screen Shot 2022-08-29 at 18 13 19

After step 5, no data is loaded. Note the URL is correctly updated to Aug 10, but no request is sent (confirmed in the network tool)
Screen Shot 2022-08-29 at 18 13 42

cc @ericboucher

### PR DESCRIPTION
**Pull request by JorgeMartinezG**
fixes #552 

**How to test:** Using cambodia
- Select a WMS Layer.**rainfall_dekad**
- Select a PointDataLayer, like Kobo or EWS.
- Verify that the Url is correctly generated

### CODE REVIEW

**Comment by wadhwamatic** [comment]
Working as expected. Thanks @JorgeMartinezG . Requested a review from @ericboucher too

File: src/utils/url-utils.ts [hunk 1]
```diff
@@ -60,28 +60,24 @@ export const useUrlHistory = () => {
...
+    if (filteredSelectedLayers === '') {
+      urlParams.delete(layerKey);
 
-    replace({ search: urlToReplace });
+      if (layerKey === UrlLayerKey.HAZARD) {
+        urlParams.delete('date');
+      }
+    } else {
+      urlParams.set(layerKey, filteredSelectedLayers);
+    }
```
Comment (ericboucher) [inline src/utils/url-utils.ts:78]: Can you add some comments to this part, please?

File: src/config/cambodia/prism.json [hunk 2]
```diff
@@ -37,9 +37,6 @@
       ],
       "early_warning": [
         "ews_remote"
-      ],
-      "flood_events": [
```
Comment (ericboucher) [inline src/config/cambodia/prism.json:41]: no more flood events?

**Review by ericboucher — APPROVED** [review]
Looks good, let's add some comments to the function and you can merge after that @JorgeMartinezG

File: src/utils/url-utils.ts [hunk 3]
```diff
@@ -60,28 +60,27 @@ export const useUrlHistory = () => {
...
       .join(',');
 
-    const urlToReplace =
-      filteredSelectedLayers === '' ? removeKey() : filteredSelectedLayers;
+    // If the list of layers is empty, remove the layerKey from the url.
+    if (filteredSelectedLayers === '') {
+      urlParams.delete(layerKey);
 
-    replace({ search: urlToReplace });
+      // For hazard layer, remove also the date.
```
Comment (ericboucher) [inline src/utils/url-utils.ts:75]: Could we simply delete the date every time when the list is empty and a date exists?

Comment (wadhwamatic) [reply src/utils/url-utils.ts:75]: > there is a possibility that a baselinelayer (admin level data) might have dates. I think there has been previous discussions about including dates in an admin level data layer

correct - https://github.com/WFP-VAM/prism-frontend/pull/555

Comment (wadhwamatic) [reply src/utils/url-utils.ts:75]: we'll need to revisit the logic once admin level dates are incorporated

**Pull request merged**